crates/biome_js_formatter/src/js/bindings/parameters.rs

```diff
@@ -223,15 +227,26 @@ pub(crate) fn should_hug_function_parameters(
                             // always true for `[a]` or `{a}`
                             AnyJsBindingPattern::JsArrayBindingPattern(_)
                             | AnyJsBindingPattern::JsObjectBindingPattern(_) => true,
-                            // only if the type parameter is an object type
+                            // if the type parameter is an object type
                             // `a: { prop: string }`
+                            // or parameter is an arrow function parameter
+                            // (a) => {}
                             AnyJsBindingPattern::AnyJsBinding(
                                 AnyJsBinding::JsIdentifierBinding(_),
-                            ) => parameter
-                                .type_annotation()
-                                .map_or(false, |type_annotation| {
-                                    matches!(type_annotation.ty(), Ok(AnyTsType::TsObjectType(_)))
-                                }),
+                            ) => {
+                                if has_parentheses && from_arrow_function {
```

Comment:
It seems that these two `bool` are always compared **together** and they must have the same value. This means that we can pass **one** `bool`, which we compute when we **call** `hug_formal_parameter`.

Reply:
done~